tracer/src/Datadog.Trace/Debugger/LineProbeResolver.cs

```diff
@@ -105,31 +106,41 @@ private Trie CreateTrieForSourceFilePaths(Assembly loadedAssembly)
             return trie;
         }
 
-        private Assembly FindAssemblyContainingFile(string sourceFileFullPath)
+        private bool TryFindAssemblyContainingFile(string sourceFileFullPath, out string sourceFileFullPathFromPdb, out Assembly assembly)
         {
             lock (_locker)
             {
-                foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+                foreach (var candidateAssembly in AppDomain.CurrentDomain.GetAssemblies())
                 {
-                    var trie = GetSourceFilePathForAssembly(assembly);
+                    var trie = GetSourceFilePathForAssembly(candidateAssembly);
+                    if (trie == null)
+                    {
+                        continue;
+                    }
 
-                    if (trie != null && trie.ContainsPrefix(GetReversePath(sourceFileFullPath)))
+                    // Because we're actually interested in matching the ending of the string rather than the beginning,
+                    // we reverse the string before inserting or querying the trie, and then reverse it again when we get a string out.
+                    var reversedPath = trie.GetStringStartingWith(GetReversePath(sourceFileFullPath));
+                    if (reversedPath != null)
                     {
-                        return assembly;
+                        sourceFileFullPathFromPdb = GetReversePath(reversedPath);
```

Comment:
Just so I understand correctly, you're calling `GetReversePath` twice (instead of using `sourceFileFullPath`) so that `sourceFileFullPathFromPdb` has the "normalized" path separators, right? Might be worth mentioning that in the comment 🙂

Reply:
Not quite. The trie is built from the PDB, so it will contains absolute paths such as:
                `"D:\build_agent\yada\yada\src\MyProject\MyFile.cs"`
...whereas what we are trying to look up is typically a path within the Git repo, such as:
                `"src/MyProject/MyFile.cs"`

Because we're actually interested in matching the ending of the string rather than the beginning,
we reverse the paths before inserting or querying the trie, and then reverse again when we get the string out. 

Does it make sense? I added this paragraph as a comment in bb274acca45b97f36f24cb60796755a9cc7f54aa to make things clearer.

Note that your comment made me think a bit further and realize that the fact that I was "normalizing" the file paths to use `Path.DirectorySeparatorChar` is incorrect and would cause an issue if the we're running on Linux while the PDB was generated on Windows (or vice versa), so I changed the implementation and added a test for it in bb274acca45b97f36f24cb60796755a9cc7f54aa.